[Microbiome] Add tutorial for MAGs building and annotation from raw reads (#6556)
galaxyproject/training-material: topics/microbiome/tutorials/mags-building/tutorial.md

  - High-quality MAGs are essential for accurate analysis. Completeness and contamination metrics are key indicators of MAG quality. High-quality MAGs ensure reliable taxonomic assignments and functional annotations, which are necessary for meaningful biological interpretations.
  - Taxonomic and functional annotation reveal the biological roles of microbes. Taxonomic assignment connects your MAGs to known microbial lineages, facilitating comparisons with existing databases. Functional annotation uncovers the metabolic, ecological, and pathological potential of microbial communities, providing insights into their biological roles and interactions.
  - Metagenomic analysis is an iterative process. It involves multiple interconnected steps, from preprocessing to functional annotation. Each step builds on the previous one, and revisiting and refining your analyses can lead to deeper insights and more accurate conclusions. As tools and databases evolve, staying updated ensures that your analyses remain cutting-edge and reliable.
edam_ontology:
  - topic_3174 # Metagenomics
  - topic_0196 # Sequence assembly
contributions:
  authorship:

>   - bebatut

Reviewer: ```suggestion
  - bebatut
    editing:
  - paulzierep
```

  funding:
  - elixir-europe
  - ifb
subtopic: metagenomics
tags:

>   - assembly

Reviewer: ```suggestion
  - assembly
  - binning
```

{: .comment}

Let's import the workflow from WorkflowHub:

{% snippet faqs/galaxy/workflows_run_wfh.md title="short-read-qc-trimming/main" wfhub_id="2025" version="1"%}

We can now launch it:

> <hands-on-title>Control raw data quality</hands-on-title>
>
> 1. Run the workflow using the following parameters
>    - {% icon param-collection %} *"Raw reads"*: `Raw reads` collection
>    - *"Qualified quality score"*: `15`
>    - *"Minimal read length"*: `15`
> 2. Inspect the MultiQC report
{: .hands_on}

> <question-title></question-title>
>
> 1. How many reads are in each dataset?
> 2. What are the length of the reads before trimming?
> 3. What are the miminum values for average bp quality score for forward and reverse reads, before and and after trimming and filtering?
>
> > <solution-title></solution-title>
> >
> > 1. 43.9 millions reads for SRR24759598 and 42.8 millions for SRR24759616
> > 2. 150 bp
> > 3. Minimum average bp quality scores given the "Sequence Quality" graph:
> > 
> >     | SRR24759598 | SRR24759616
> >    --- | --- | ---
> >    Forward (Read 1) - Before | 34.5 | 34.6
> >    Forward (Read 1) - After | 34.5 | 34.6
> >    Reverse (Read 2) - Before | 33.4 | 32.7

> > >    Reverse (Read 2) - After | 33.4 | 32.7

Reviewer: I guess they deposited trimmed reads

7. **Taxonomic Assignment**: Classify the MAGs using **GTDB-Tk** ({% cite Chaumeil2019 %}), aligning them with the Genome Taxonomy Database for accurate taxonomic placement.
8. **Functional Annotation**: Identify and annotate genes within our MAGs using **Bakta** ({% cite Schwengers_2021 %}), revealing their functional potential and biological roles.

All results are consolidated into a single **MultiQC** ({% cite ewels2016multiqc %}) report, allowing us to easily visualize and interpret the quality and characteristics of our MAGs.

Let's import the workflow from WorkflowHub:

{% snippet faqs/galaxy/workflows_run_wfh.md title="mags-building/main" wfhub_id="1352" version="4"%}

> <hands-on-title>Build, refine, and annotate MAGs</hands-on-title>
>
> 1. Run the workflow using the following parameters
>    - *"Trimmed reads from grouped samples"*: `Reads without bee and human reads`
>    - *"Trimmed reads"*: `Reads without bee and human reads`
>    - *"Choose Assembler"*: `MEGAHIT`
>
>      > <comment-title></comment-title>
>      > metaSPAdes is an alternative assembler.
>      >

> >      > MEGAHIT is less computationally intensive and generate higher quality single and shorter contigs but shorter. metaSPAdes is very computationally intensive, but generates longer/more complete assemblies.

Reviewer: strange sentence, and maybe point to https://doi.org/10.1093/bib/bbad087 for a benchmark

- **Data Limitations:** Minor community members may be underrepresented, leading to incomplete or fragmented assemblies.

To address these challenges, a variety of **metagenomic assemblers** have been developed, including **metaSPAdes** ({% cite Nurk_2017 %}) and **MEGAHIT** ({% cite Li_2015 %}). Each assembler has unique computational characteristics, and their performance can vary depending on the microbiome being studied. As demonstrated by the **Critical Assessment of Metagenome Interpretation (CAMI) initiative** ({% cite sczyrba2017 %}, {% cite meyer2021 %}, {% cite meyer2022 %}), the choice of assembler often depends on the specific goals of the analysis, such as maximizing contiguity, accuracy, or computational efficiency. Selecting the right tool is essential for achieving high-quality assemblies tailored to your research objectives.

> <comment-title></comment-title>
> Learn more about metagenomic assembly, algorithms and the tools in our dedicated tutorial: [**Assembly of metagenomic sequencing data**]({% link topics/microbiome/tutorials/metagenomics-assembly/tutorial.md %})
{: .comment}

> In this workflow, we used **MEGAHIT**, but **metaSPAdes** is also proposed as an alternative assembler.

Reviewer: ```suggestion
In this workflow execution, we used **MEGAHIT**, but **metaSPAdes** is also an option as an alternative assembler in the workflow.
```

   > > 2. L50 valueS:
   > >   - 11,641 for SRR24759598
   > >   - 12,941 for SRR24759616.
   > > 3. The N50 value is higher for SRR24759598 (2,213 bp) compared to SRR24759616 (1,751 bp), indicating that the assembly for SRR24759598 is slightly more contiguous, with longer contigs covering half of the total assembly length. However, the L50 value is lower for SRR24759598 (11,641) than for SRR24759616 (12,941), meaning fewer contigs are needed to reach 50% of the total assembly length in SRR24759598. This suggests that while both assemblies are fragmented, SRR24759598 has a modestly better contiguity overall. The differences in these metrics may reflect variations in microbial diversity, sequencing depth, or assembly performance between the two samples.
   > {: .solution}
   >
   {: .question}

- **Read mapping**: results of the mapping of the raw reads on the contigs

    > <question-title></question-title>
    >
    > 1. What is the % of SRR24759598 reads mapped to SRR24759598 contigs? And for SRR24759616?
    > 2. What is the percentage of reads used to build the assemblies for SRR24759598? and SRR24759616?
    >
    > > <solution-title></solution-title>
    > >

>     > > 1. % of mapped reads:

Reviewer: ```suggestion
    > > 1. percentage (%) of mapped reads:
```

Binning relies on a combination of **sequence composition, coverage, and similarity** to group contigs into bins. These bins ideally represent the genomes of distinct microorganisms present in the sample. However, the process is inherently challenging due to:
- **Genomic complexity**, such as repeated sequences and strain-level variation.
- **Uneven sequencing coverage** across different microbial species.
- **Contamination and chimeras**, which can obscure the boundaries between genomes.

Numerous algorithms have been developed for metagenomic binning, each with unique strengths and limitations. The choice of tool often depends on the dataset's characteristics and the research objectives. Benchmark studies (e.g., {% cite sczyrba2017 %}, {% cite meyer2021 %}, {% cite meyer2022 %}) provide practical guidance on selecting the most suitable binner for specific environments and datasets.

> <comment-title></comment-title>
> Learn more about metagenomic binning and available tools in our dedicated tutorial: [**Binning of metagenomic sequencing data**]({% link topics/microbiome/tutorials/metagenomics-binning/tutorial.md %})
{: .comment}

> A widely adopted strategy is to **use multiple binning tools** to maximize accuracy, followed by **bin refinement** to consolidate and improve the results. This approach leverages the strengths of different algorithms, producing a more robust and reliable set of bins.

Reviewer: ```suggestion
A widely adopted strategy is to **use multiple binning tools** to profit from their individual advantages and methods, followed by **bin refinement** to consolidate and improve the results. This approach leverages the strengths of different algorithms, producing a more robust and reliable set of bins.
```
